Question 2

```diff
@@ -0,0 +1,305 @@
+            recoverI = 0
+    for i in range(I):
+        if random.random() < g:
+            recoverI += 1
+    
+    S -= newI
+    I += (newI - recoverI)
+    R += recoverI
+    
+    #Append
+    sList.append(S)
+    iList.append(I)
+    rList.append(R)
+    newIList.append(newI)
+    
+    print('t', t)
+    t += 1
+#Print
+print('sList', sList)
+print('iList', iList)
+print('rList', rList)
+print('newIList', newIList)
+
+#values
+beta=0.00005
+gamma=0.1
+#time (days)
+ t = np.linspace(0, 500, 500)
+ while I > 0:
+    newI = 0
+        for i in range(S):
+      
+        if random.random() < b*(I/N):
+            newI += 1
+            recoverI = 0
+    for i in range(I):
+        if random.random() < g:
+            recoverI += 1
+    
+    S -= newI
+    I += (newI - recoverI)
+    R += recoverI
+    
+    #Append
+    sList.append(S)
+    iList.append(I)
+    rList.append(R)
+    newIList.append(newI)
+    
+    print('t', t)
+    t += 1
+#Print
+print('sList', sList)
+print('iList', iList)
+print('rList', rList)
+print('newIList', newIList)
+
+#values
+beta=0.0001
+gamma=0.05
+#time (days)
+ t = np.linspace(0, 500, 500)
+ while I > 0:
+    newI = 0
+        for i in range(S):
+      
+        if random.random() < b*(I/N):
+            newI += 1
+            recoverI = 0
+    for i in range(I):
+        if random.random() < g:
+            recoverI += 1
+    
+    S -= newI
+    I += (newI - recoverI)
+    R += recoverI
+    
+    #Append
+    sList.append(S)
+    iList.append(I)
+    rList.append(R)
+    newIList.append(newI)
+    
+    print('t', t)
+    t += 1
+#Print
+print('sList', sList)
+print('iList', iList)
+print('rList', rList)
+print('newIList', newIList)
+
+#values
+beta=0.0002
+gamma=0.05
+#time (days)
+ t = np.linspace(0, 500, 500)
+ while I > 0:
+    newI = 0
+        for i in range(S):
+      
+        if random.random() < b*(I/N):
+            newI += 1
+            recoverI = 0
+    for i in range(I):
+        if random.random() < g:
+            recoverI += 1
+    
+    S -= newI
+    I += (newI - recoverI)
+    R += recoverI
+    
+    #Append
+    sList.append(S)
+    iList.append(I)
+    rList.append(R)
+    newIList.append(newI)
+    
+    print('t', t)
+    t += 1
+#Print
+print('sList', sList)
+print('iList', iList)
+print('rList', rList)
+print('newIList', newIList)
+
+#values
+beta=0.0001
+gamma=0.06
+#time (days)
+ t = np.linspace(0, 500, 500)
+ while I > 0:
+    newI = 0
+        for i in range(S):
+      
+        if random.random() < b*(I/N):
+            newI += 1
+            recoverI = 0
+    for i in range(I):
+        if random.random() < g:
+            recoverI += 1
+    
+    S -= newI
+    I += (newI - recoverI)
+    R += recoverI
+    
+    #Append
+    sList.append(S)
+    iList.append(I)
+    rList.append(R)
+    newIList.append(newI)
+    
+    print('t', t)
+    t += 1
+#Print
+print('sList', sList)
+print('iList', iList)
+print('rList', rList)
+print('newIList', newIList)
+
```

Comment:
Define the function:
def simSIR(y,t,beta,gamma):
    S=y[0]
    I=y[1]
    R=y[2]
    dSdt=-beta*I*S
    dIdt=beta*I*S-gamma*I
    dRdt=gamma*I
    return [dSdt,dIdt,dRdt]

Define the parameters:
y0=[999,1,0]
times=range(0,500)

betas=numpy.array([0.0005,0.005,0.0001,0.00005,0.0001,0.0002,0.0001])
gammas=numpy.array([0.05,0.5,0.1,0.1,0.05,0.05,0.06])

Make a new dataframe to store the parameters: 
storeSIR=pandas.DataFrame({"beta":betas,"gamma":gammas,"R0":betas*sum(y0)/gammas,"maxIncidence":0,"maxPrevalence":0,"percentAffected":0})

Using the for loop to simulate: 
for i in range(0,len(betas)):
    pars=(betas[i],gammas[i])
    sim=si.odeint(func=simSIR,y0=y0,t=times,args=pars)
    storeSIR.iloc[i,4]=numpy.max(sim[:,1]/numpy.sum(sim,axis=1))
    storeSIR.iloc[i,3]=numpy.max(sim[1:len(sim),1]-sim[0:(len(sim)-1),1])
    storeSIR.iloc[i,5]=numpy.sum(sim[len(sim)-1,1:3])/numpy.sum(sim[len(sim)-1,:])*100

-0.5